python/lsst/sconsUtils/state.py

```diff
@@ -128,6 +130,11 @@ def _initEnvironment():
                 ourEnv[varname] = os.environ[varname]
                 ourEnv[k] = os.environ[k]
 
+    # add <build root>/ups directory to the configuration search path
+    # this allows the .cfg file for the package being built to be found without
+    # requiring <product name>_DIR to be in the env
+    cfgPath.append(os.path.join(SCons.Script.Dir('#').abspath, 'ups'))
```

Comment:
# Shouldn't scons only be looking for _DIR variables associated with entries in the table file of the application being built?
You mean, associated with entries in any table file included recursively.